File: Palindrome_Number/Palindrome_Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x==0){
            return true;
        }else if(x<0){
            return false;
        }
        int num = -1;
        int tmp = x;
        while(tmp){
            tmp /= 10;
            num++;
        }
        if(num==9 && x%10>2){
            return false;
        }
        int x_rev = 0;
        int x_ori = x;
        while(x_ori){
            x_rev += (x_ori%10) * pow(10, num);
            num--;
            x_ori /= 10;
        }
        if(x == x_rev){
            return true;
        }
        return false;
    }
};

File: Palindrome_Number/Palindrome_Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if(x<0):
            return False
        else:
            reverse = 0
            x_cmp = x
            while(x_cmp):
                reverse = reverse*10 + x_cmp%10;
                x_cmp /= 10;
        return x == reverse
